GPflow/vgp.py

```diff
@@ -100,7 +104,10 @@ def build_predict(self, Xnew):
             A = K + tf.diag(1./tf.square(b))
             L = tf.cholesky(A)
             LiKx = tf.matrix_triangular_solve(L, tf.transpose(Kx), lower=True)
-            f_var.append( Kd - tf.reduce_sum(tf.square(LiKx),0) )
+            if full_cov:
+                f_var.append( self.kern.K(Xnew)- tf.matmul(tf.transpose(LiKx),LiKx) )
```

Comment:
Here we're looping over the number of latent functions (i.e. over the columns of Y). We're going to return the variances as a NxD matrix (N test points, D functions) or as a NxNxD tensor of 'full' covariances. We want to stack them, not add them.